pkg/antctl/raw/traceflow/command.go

```diff
@@ -137,22 +138,29 @@ func runE(cmd *cobra.Command, _ []string) error {
 		return nil
 	}
 
-	if err := wait.Poll(1*time.Second, 15*time.Second, func() (bool, error) {
-		tf, err := client.OpsV1alpha1().Traceflows().Get(context.TODO(), tf.Name, metav1.GetOptions{})
+	var res *v1alpha1.Traceflow
+	err = wait.Poll(1*time.Second, 15*time.Second, func() (bool, error) {
+		res, err = client.OpsV1alpha1().Traceflows().Get(context.TODO(), tf.Name, metav1.GetOptions{})
 		if err != nil {
 			return false, err
 		}
-		if tf.Status.Phase != v1alpha1.Succeeded {
+		if res.Status.Phase != v1alpha1.Succeeded && res.Status.Phase != v1alpha1.Failed {
 			return false, nil
 		}
-		if err := output(tf); err != nil {
-			return false, fmt.Errorf("error when outputing result: %w", err)
-		}
 		return true, nil
-	}); err != nil {
+	})
+	if err == wait.ErrWaitTimeout {
+		// Still output the Traceflow results if any.
+		if res == nil || len(res.Status.Results) == 0 {
+			return errors.New("timeout waiting for Traceflow results")
+		}
```

Comment:
shouldn't we still return a timeout  error eventually after outputting the (partial?) result?

Reply:
We can do that too. In your mind should antctl exit with an error in that case?

Reply:
yes I believe so

Reply:
Ok. Updated the code.

Reply:
`if err := output(res); err != nil {` should not override err?

Reply:
Oh, sorry, you're right... Become careless after holiday :(